cmake/toolchain/zephyr/0.11/generic.cmake

```diff
@@ -6,8 +6,27 @@ set(COMPILER gcc)
 set(LINKER ld)
 set(BINTOOLS gnu)
 
-set(CROSS_COMPILE_TARGET x86_64-${TOOLCHAIN_VENDOR}-elf)
-set(SYSROOT_TARGET       x86_64-${TOOLCHAIN_VENDOR}-elf)
+# Find some toolchain that is distributed with this particular SDK
+
+foreach(target
```

Comment:
Can we build this list based on dirs in ${ZEPHYR_SDK_INSTALL_DIR} and ${ZEPHYR_SDK_INSTALL_DIR}/* (for xtensa) that are either *-zephyr-elf or *-zephyr-eabi?

Reply:
I wasn't able to find a way to get "file(GLOB_RECURSE" to do this.

Any chance we could align xtensa to behave like the other platforms?

We don't support multiple xtensa toolchains in a single zephyr-sdk do we?

Reply:
@dcpleung can hopefully chime in on the xtensa side of things.  But we have to deal with structure as it is right now because of shipped SDK 0.11.x releases.

Reply:
> We don't support multiple xtensa toolchains in a single zephyr-sdk do we?

what do you mean by this?

Reply:
Something like this works on my computer:

```
file(GLOB toolchain_paths
    LIST_DIRECTORIES true
    ${TOOLCHAIN_HOME}/*/x86_64-zephyr-elf
    ${TOOLCHAIN_HOME}/*/aarch64-zephyr-elf
    ${TOOLCHAIN_HOME}/*/arm-zephyr-eabi
    ${TOOLCHAIN_HOME}/*/nios2-zephyr-elf
    ${TOOLCHAIN_HOME}/*/riscv64-zephyr-elf
    ${TOOLCHAIN_HOME}/*/mipsel-zephyr-elf
    ${TOOLCHAIN_HOME}/*/arc-zephyr-elf
    ${TOOLCHAIN_HOME}/xtensa/*/*/xtensa-zephyr-elf
    )

if(toolchain_paths)
  list(GET  toolchain_paths 0 some_toolchain_path)

  get_filename_component(one_toolchain "${some_toolchain_path}" NAME)

  set(CROSS_COMPILE_TARGET ${one_toolchain})
  set(SYSROOT_TARGET	     ${one_toolchain})
endif()
```

Reply:
If we only support placing one xtensa toolchain in a zephyr-sdk, then we wouldn't need to place them in `sdk/xtensa/<variant>/<compiler_target>`.

We could place them in 

`sdk/<compiler_target>`

just like all the other arch's. And we wouldn't have any problems.

Reply:
> file(GLOB toolchain_paths
>     LIST_DIRECTORIES true
>     ${TOOLCHAIN_HOME}/xtensa/*/xtensa-zephyr-elf
>     ${TOOLCHAIN_HOME}/x86_64-zephyr-elf
>     ${TOOLCHAIN_HOME}/aarch64-zephyr-elf
>     ${TOOLCHAIN_HOME}/arm-zephyr-eabi
>     ${TOOLCHAIN_HOME}/nios2-zephyr-elf
>     ${TOOLCHAIN_HOME}/riscv64-zephyr-elf
>     ${TOOLCHAIN_HOME}/mipsel-zephyr-elf
>     ${TOOLCHAIN_HOME}/arc-zephyr-elf
>     )


Can you make this:
```

file(GLOB toolchain_paths
    LIST_DIRECTORIES true
    ${TOOLCHAIN_HOME}/xtensa/*/*-zephyr-elf
    ${TOOLCHAIN_HOME}/*-zephyr-elf
    ${TOOLCHAIN_HOME}/*-zephyr-eabi
    )
```

Reply:
@dcpleung : I'm getting

```
CMake Error at /home/sebo/ncs/zephyr/cmake/extensions.cmake:1479 (message):
  No such file or directory: LIBGCC_FILE_NAME: ''
Call Stack (most recent call first):
  /home/sebo/ncs/zephyr/cmake/compiler/gcc/target.cmake:69 (assert_exists)
  /home/sebo/ncs/zephyr/cmake/target_toolchain.cmake:49 (include)
  /home/sebo/ncs/zephyr/cmake/app/boilerplate.cmake:473 (include)
  CMakeLists.txt:5 (include)

```

with your patch.

On hello world and intel_s1000_crb.

Reply:
Which board does xtensa_intel_apl_adsp work with? I can't find a board by that name.

Reply:
I got your patch to work with the s1000 toolchain and board now ...

Reply:
I have tested and pushed @dcpleung 's suggestion along with @galak 's suggestion.

Reply:
`intel_apl_adsp` is the SoC for board `up_squared_adsp`.